chat2_server.lua

```diff
@@ -26,48 +33,36 @@ function useDefaultOutput(bool)
 end
 
 function onChatSendMessage(message)
-  local sender = client
-  local nickname = getPlayerName(sender)
-
   if type(message) ~= "string" or utf8.len(message) < minLength or utf8.len(message) > maxLength then
     return
   end
 
+  local sender = client
+
   if utf8.sub(message, 0, 1) == "/" then
     handleCommand(sender, message)
     return
   end
 
   if not isDefaultOutput then
-    triggerEvent("onPlayerChat2", root, sender, message)
+    triggerEvent("onPlayerChat2", rootPlayers, sender, message)
     return
   end
 
+  local nickname = getPlayerName(sender)
+
   for _, player in ipairs(getElementsByType("player")) do
     local text = string.format("%s#ffffff: %s", nickname, message)
-    output(player, text)
-    outputServerLog(text)
   end
+  output(player, text)
```

Comment:
I think you wanted to execute `output` inside of a loop but `outputServerLog` outside